# Awesome Linked Reads

This is a collection of tools and resources for analysis and processing of linked-reads. 

## Tools
Name|Category|Description|Status
---|---|:---|-----
[Aquila](https://github.com/maiziex/Aquila)|assembly, pipeline|Diploid personal genome assembly and comprehensive variant detection based on linked-reads|![GitHub last commit](https://img.shields.io/github/last-commit/maiziex/Aquila)
[Aquila_stLFR](https://github.com/maiziex/Aquila_stLFR)|assembly, pipeline|Human haplotype-resolved assembly and variant detection for stLFR, hybrid assembly for linked-reads|![GitHub last commit](https://img.shields.io/github/last-commit/maiziex/Aquila_stLFR)
[Ariadne](https://github.com/lauren-mak/ariadne)|assembly, metagenomics|de Bruijn graph-based program for barcoded read deconvolution|![GitHub last commit](https://img.shields.io/github/last-commit/lauren-mak/ariadne)
[arcs](https://github.com/bcgsc/arcs)|assembly|Scaffold genome sequence assemblies. |![GitHub last commit](https://img.shields.io/github/last-commit/bcgsc/arcs)
[Athena](https://github.com/abishara/athena_meta)|assembly, metagenomics|Read cloud assembler for metagenomes|![GitHub last commit](https://img.shields.io/github/last-commit/abishara/athena_meta)
[BarCrawler](https://github.com/J35P312/BarCrawler)|qc|QC package for 10X genomics barcoded reads.|![GitHub last commit](https://img.shields.io/github/last-commit/J35P312/BarCrawler)
[bxtools](https://github.com/walaj/bxtools)|toolkit| Tools for analyzing mapped 10x data|![GitHub last commit](https://img.shields.io/github/last-commit/walaj/bxtools)
[EMerAld (EMA)](https://github.com/arshajii/ema)|mapping|Preforms barcode-aware alignment of linked reads. Also does preprocessing of 10x Genomics data.|![GitHub last commit](https://img.shields.io/github/last-commit/arshajii/ema)
[grocsvs](https://github.com/grocsvs/grocsvs)|structural variants|Genome-wide reconstruction of complex structural variants|![GitHub last commit](https://img.shields.io/github/last-commit/grocsvs/grocsvs)
[HapCUT2](https://github.com/vibansal/HapCUT2)|phasing|Phasing of barcode linked reads|![GitHub last commit](https://img.shields.io/github/last-commit/vibansal/HapCUT2)
[HapTree-X](https://github.com/0xTCG/haptreex)|phasing|Haplotype phaser for next-generation sequencing data|![GitHub last commit](https://img.shields.io/github/last-commit/0xTCG/haptreex)
[Lariat](https://github.com/10XGenomics/lariat)|mapping|Linked-Read Alignment Tool|![GitHub last commit](https://img.shields.io/github/last-commit/10XGenomics/lariat)
[LEVIATHAN](https://github.com/morispi/LEVIATHAN)|structural variants|Linked-reads based structural variant caller with barcode indexing|![GitHub last commit](https://img.shields.io/github/last-commit/morispi/LEVIATHAN)
[LinkedSV](https://github.com/WGLab/LinkedSV)|structural variants|Structural variant caller for linked-read sequencing data|![GitHub last commit](https://img.shields.io/github/last-commit/WGLab/LinkedSV)
[LongRanger](https://github.com/10XGenomics/longranger)|pipeline|Pipeline for alignment, variant calling, phasing, and ptructural variant calling|![GitHub last commit](https://img.shields.io/github/last-commit/10XGenomics/longranger)
[LRez](https://github.com/morispi/LRez)|toolkit|Standalone tool and library allowing to work with barcoded linked-reads|![GitHub last commit](https://img.shields.io/github/last-commit/morispi/LRez)
[LRTK-SIM](https://github.com/zhanglu295/LRTK-SIM)|simulation|A program to simulate linked reads sequencing from 10X Chromium System|![GitHub last commit](https://img.shields.io/github/last-commit/zhanglu295/LRTK-SIM)
[LRSIM](https://github.com/aquaskyline/LRSIM)|simulation|A simulator for linked reads|![GitHub last commit](https://img.shields.io/github/last-commit/aquaskyline/LRSIM)
[Minerva](https://github.com/dcdanko/minerva_barcode_deconvolution)|assembly|Sort Linked Read DNA Into Fragment Specific Clusters|![GitHub last commit](https://img.shields.io/github/last-commit/dcdanko/minerva_barcode_deconvolution)
[NAIBR](https://github.com/raphael-group/NAIBR)|structural variants|Identifies novel adjacencies created by structural variation events such as deletions, duplications, inversions, and complex rearrangements|![GitHub last commit](https://img.shields.io/github/last-commit/raphael-group/NAIBR)
[Novel-X](https://github.com/1dayac/Novel-X)|structural variants|Novel insertion detection with 10X reads|![GitHub last commit](https://img.shields.io/github/last-commit/1dayac/Novel-X)
[Pyslr](https://github.com/bcgsc/physlr)|assembly|Construct a Physical Map from Linked Reads|![GitHub last commit](https://img.shields.io/github/last-commit/bcgsc/physlr)
[Samovar](https://github.com/cdarby/samovar)|variant calling|Somatic (mosaic) SNV caller for 10X Genomics data using random forest classification and feature-based filters|![GitHub last commit](https://img.shields.io/github/last-commit/cdarby/samovar)
[scaff10x](https://sourceforge.net/projects/phusion2/files/scaff10x/)|assembly|Scaffolding assemblies using linked read data |
[stLFRdenovo](https://github.com/BGI-biotools/stLFRdenovo)|assebly|De Novo assembly pipeline to deal with barcoded reads. It is based on Supernova, with a fastq parsing and sorting module constumized for stLFR data.|![GitHub last commit](https://img.shields.io/github/last-commit/BGI-biotools/stLFRdenovo) 
[stLFRsv](https://github.com/BGI-biotools/stLFRsv)|structural variants|Structure variation(SV) pipeline for stLFR co-barcode reads|![GitHub last commit](https://img.shields.io/github/last-commit/BGI-biotools/stLFRsv)
[SuperNova](https://github.com/10XGenomics/supernova)|assembly|10x Genomics Linked-Read Diploid De Novo Assembler|![GitHub last commit](https://img.shields.io/github/last-commit/10XGenomics/supernova)
[SVenX](https://github.com/vborjesson/SVenX)|structural variants|Pipeline for SV detection using 10X genomics data|![GitHub last commit](https://img.shields.io/github/last-commit/vborjesson/SVenX)
[tenx_utils](https://github.com/friend1ws/tenx_utils)|toolkit|Utility functions for 10x data|![GitHub last commit](https://img.shields.io/github/last-commit/friend1ws/tenx_utils)
[Tigmint](https://github.com/bcgsc/tigmint)|assembly|Correct misassemblies using Linked Reads|![GitHub last commit](https://img.shields.io/github/last-commit/bcgsc/tigmint)
[Topsorter](https://github.com/hanfang/Topsorter)|structural variants, qc|Graphic assement of structural variants|![GitHub last commit](https://img.shields.io/github/last-commit/hanfang/Topsorter)
[Valor](https://github.com/BilkentCompGen/valor)|structural variants|Variation discovery using long range information in linked-reads|![GitHub last commit](https://img.shields.io/github/last-commit/BilkentCompGen/valor)
[ZoomX](https://bitbucket.org/charade/zoomx/)|structural variants|Single Molecule Based Rearrangement Analysis with Linked Read Sequencing |


## Linked Read Platforms

### 10x Genomics Chromium / GemCode
10x Genomics linked-read technology comes in two versions; the older [GemCode (v1)](https://doi.org/10.1038/nbt.3432) and more recent [Chromium Genome (v2)]( https://doi.org/10.1101/gr.234443.118>). Long DNA fragments are combined in droplets with barcode-containing gel-beads to create GEMs ((Gel Bead-In EMulsions). The fragments are amplified and barcoded using a combination of free random hexamers and barcode-linked random hexamers from the gel beads. Following this barcoded fragments are recovered and fragments before ligation of 3' sequencing adaptor. Libraries are sequenced using Illumina Sequencing. The commercial version of the technology is [currently discontinued](https://www.10xgenomics.com/products/linked-reads).

### TELL-Seq
TELL-seq is based on the technology from [Chen et al. 2020](https://doi.org/10.1101/gr.260380.119) and is commercially available from the company [Universal Sequencing](https://www.universalsequencing.com/). The method uses clonaly barcode beads with attacted tagmentases to cut and barcode individual long DNA fragments in solution. A second tagmentation is also preformed in solution to introduce a second adaptor. The library is sequenced using Illumina sequencing with special setup to sequence the barcode as index 1.

### stLFR
stLFR (single-tube long fragment read) is based on the technology described in [Wang et al. 2019](https://doi.org/10.1101/gr.245126.118) and is commercially available from [MGI](https://en.mgi-tech.com/products/reagents_info/18/). The technology uses tagmentation to individually cut-and-hold long DNA fragments in solution. The tagmentase-DNA complex is then hybridized and individual wrapped around barcoded beads through the adaptor introduced by the tagmentation. The barcode is then ligated to each subfragment before recovery and final library prepration. Sequencing is preformed on the DNBSEQ platfroms.

### DBS
Droplet Barcode Sequencing (DBS) is based on the technology described in [Redin el al. 2019](https://doi.org/10.1038/s41598-019-54446-x). Long DNA fragments are subjected to tagmentation using Tn5-covered beads to cut, tag and wrap the fragment around the beads. The DNA-wrapped beads are then used in emmulsion PCR along with barcoded oligo. Within each emmulsion droplet the barcode and tagged fragments are amplified independently and then linked using overlap-extension. Barcode-linked fragments are recovered and indexed for Illumina sequencing.

### CPT-seq
Technologies based [Amini et al. 2014](https://doi.org/10.1038/ng.3119) and the follow-up CPTv2-seq from [Zhang et al. 2017](https://doi.org/10.1038/nbt.3897). These technologies were developed by Illumina but are not commercially available.

### Haplotagging
Haplotagging is based on the technology presented in [Meier et al. 2021](https://doi.org/10.1073/pnas.2015005118). The technology uses barcoded beads
covered with Tn5 tagmentase to cut and barcode individual long DNA fragments in solution. The beads are coated in a combination of two barcodes AB and CB
that become inserted at the 5' and 3' of each cut fragment. Barcodes are combinatorialy generated with about 85 million possible combinations in total. 

## Credits

Inspired by the collection in [Awesome-10x-genomics](https://github.com/johandahlberg/awesome-10x-genomics/).

